crypto/digestible/derive/test/tests/behavior.rs

```diff
@@ -166,108 +291,772 @@ fn generic_foo2() {
         b: String::from("str2"),
     };
 
-    let mut hasher = Tester::new();
-    arg.digest(&mut hasher);
-
-    let expected: Vec<Vec<u8>> = vec![
-        b"GenericFoo".to_vec(),
-        b"< X : Digestible >".to_vec(),
-        b"alloc::string::String".to_vec(),
-        b"a".to_vec(),
-        (4 as usize).to_le_bytes().to_vec(),
-        b"str1".to_vec(),
-        b"b".to_vec(),
-        (4 as usize).to_le_bytes().to_vec(),
-        b"str2".to_vec(),
-    ];
-
-    assert_eq!(hasher.args, expected);
+    let expected_ast = ASTNode::from(ASTAggregate {
+        context: b"genfoo2",
+        name: b"GenericFoo".to_vec(),
+        elems: vec![
+            ASTNode::from(ASTPrimitive {
+                context: b"a",
+                type_name: b"str",
+                data: "str1".as_bytes().to_vec(),
+            }),
+            ASTNode::from(ASTPrimitive {
+                context: b"b",
+                type_name: b"str",
+                data: "str2".as_bytes().to_vec(),
+            }),
+        ],
+        is_completed: true,
+    });
+    digestible_test_case_ast("genfoo2", &arg, expected_ast.clone());
+    assert_eq!(
+        arg.digest32::<MerlinTranscript>(b"genfoo2"),
+        [
+            73, 146, 142, 113, 190, 93, 124, 188, 118, 21, 72, 51, 19, 101, 141, 43, 101, 207, 234,
+            146, 65, 49, 191, 0, 40, 215, 129, 168, 55, 80, 189, 247
+        ]
+    );
+
+    let arg2 = GenericFoo {
+        a: Some(String::from("str1")),
+        b: Some(String::from("str2")),
+    };
+
+    digestible_test_case_ast("genfoo2", &arg2, expected_ast);
+    assert_eq!(
+        arg2.digest32::<MerlinTranscript>(b"genfoo2"),
+        [
+            73, 146, 142, 113, 190, 93, 124, 188, 118, 21, 72, 51, 19, 101, 141, 43, 101, 207, 234,
+            146, 65, 49, 191, 0, 40, 215, 129, 168, 55, 80, 189, 247
+        ]
+    );
+}
+
+// An example structure
+#[derive(Digestible)]
+struct Thing {
```

Comment:
It looks like these types and the following tests are all about schema evolution. If they are self-contained, it might help to move them into a separate file, like tests/schema_evolution.rs